TSPL.docc/ReferenceManual/Attributes.md

```diff
@@ -345,6 +345,68 @@ struct MyStruct {
   ```
 -->
 
+### backDeployed
+
+Apply this attribute to a function, method, subscript, or computed property
+to include a copy of the symbol's implementation
+in programs that call or access the symbol.
+You use this attribute to annotate symbols that ship as part of an SDK
+when they can be made available retroactively
+by including a copy of their implementation in programs that access them.
+Copying the implementation is also known as *emitting into the client*.
+
+This attribute takes a `before:` argument,
+specifying the first version of platforms that provide this symbol.
+Unlike the `available` attribute,
+the list platforms can't contain an asterisk (`*`) to refer to all platforms.
+For example, consider a library that contains the following code:
+
+```swift
+@available(iOS 16, *)
+@backDeployed(before: iOS 17)
+func someFunction() { /* ... */ }
+```
+
+In the example above,
+the iOS SDK provides `someFunction()` starting in version 17.
+In addition,
+`someFunction()` is available on iOS 16 using back deployment.
+
+When compiling code that calls this function,
+Swift inserts a layer of indirection that find the function's implementation.
+If the code is run using a version of the SDK that includes this function,
+the SDK's implementation is used.
+Otherwise, the copy included in the caller is used.
+In the example above,
+calling `someFunction()` uses the implementation from the SDK
+when running on iOS 17 or later,
+and when running on iOS 16
+it uses the copy of `someFunction()` that's included in the caller.
+
+> Note:
+> When the caller's minimum deployment target
+> is the same as or greater than
+> the first version of the SDK that includes the symbol,
+> the compiler can optimize away the runtime check
+> and call the SDK's implementation directly.
+
+<!--
+XXX TR: The intro to the SE proposal suggests that it can also omit
+the emit-into-client copy of the implementation
+when the minimum deployment target is high enough.
+Is that always the case?
+Or does that depend on other optimizations like dead-code stripping?
+-->
```

Comment:
In straightforward cases where you just call the `@backDeployed` function from some app code, this is a guaranteed optimization. However, there are some more complex cases, like when some other `@inlinable` API makes a call to a `@backDeployed` API, where we have to rely on the optimizer to inline the thunk implementation, constant fold the redundant OS availability check that was inlined from the thunk, and then finally remove the dead fall back function that is no longer called anywhere. I think in practice this is all fairly reliable for optimized builds, but we probably ought not guarantee it since there are a lot of moving parts involved and optimization inhibitors are sometimes hard to predict.

Reply:
Thanks — added in d4b5c2d5.